File: PersianCalendar/src/main/java/com/byagowi/persiancalendar/Widget4x2.java
package com.byagowi.persiancalendar;

import android.appwidget.AppWidgetProvider;
import android.content.Context;
import android.content.Intent;

import com.byagowi.persiancalendar.util.UpdateUtils;
import com.byagowi.persiancalendar.util.Utils;

public class Widget4x2 extends AppWidgetProvider {
    @Override
    public void onReceive(Context context, Intent intent) {
        Utils.startEitherServiceOrWorker(context);
        UpdateUtils.update(context, false);
    }
}

File: PersianCalendar/src/main/java/com/byagowi/persiancalendar/Widget4x2.kt
package com.byagowi.persiancalendar

import android.appwidget.AppWidgetProvider
import android.content.Context
import android.content.Intent

import com.byagowi.persiancalendar.utils.UpdateUtils
import com.byagowi.persiancalendar.utils.Utils

class Widget4x2 : AppWidgetProvider() {
    override fun onReceive(context: Context, intent: Intent) {
        Utils.startEitherServiceOrWorker(context)
        UpdateUtils.update(context, false)
    }
}
